### ISSUE TITLE
Issue with as3 settings on DSC devices that synchronize config

### ISSUE THREAD

**Issue by m-kratochvil** [F5Networks/f5-appsvcs-extension#525]
### Environment
 * Application Services Version:
```
  "version": "3.32.0",
  "release": "4",
  "schemaCurrent": "3.32.0",
  "schemaMinimum": "3.0.0"
```

 * BIG-IP Version:
```
  Product     BIG-IP
  Version     15.1.3
  Build       0.21.11
  Edition     Engineering Hotfix
```

### Summary
The AS3 settings have two components:
- the API endpoint `/mgmt/shared/appsvcs/settings`
- the internal data-group `/Common/appsvcs/settings`

I ran into an issue during AS3 upgrade using Ansible (URI module), targeting two devices that are synchronizing config between each other automatically.

I believe that the issue is that the data-group `/Common/appsvcs/settings` synchronizes between the devices, but the API endpoint settings do not. Ansible configures both devices in parallel. When the first device is configured, both the data-group and the API endpoint are correctly updated. The data-group is synchronized to the peer, but the API settings are not. When Ansible tries to configure the second device, it fails with following error:

```
"json": {
        "code": 500,
        "message": "Command failed: tmsh -a delete ltm data-group internal /Common/appsvcs/settings\n01020036:3: The requested value list (/Common/appsvcs/settings) was not found.\n"
}
```
with corresponding error in `/var/log/restnoded/restnoded.log`:
```
severe: [appsvcs] {"message":"Unable to update AS3 settings due to:\nError: Command failed: tmsh -a delete ltm data-group internal /Common/appsvcs/settings\n01020036:3: The requested value list (/Common/appsvcs/settings) was not found.\n","level":"error"}
```

This is most likely because the config sync operation from the first device is happening at that same moment, and the data-group is being replaced (deleted/created).

### Steps To Reproduce
To reproduce the issue, use the Ansible URI module to update the AS3 settings - on a pair of devices that have auto-sync configuration enabled.

For example, my task looks like:
```
- name: Modify AS3 settings.
  ansible.builtin.uri:
    url: https://{{ ansible_host }}/mgmt/shared/appsvcs/settings
    method: POST
    body_format: json
    body: "{{ as3_settings }}"
    url_username: <user>
    url_password: <password>
    use_proxy: no
    validate_certs: no
```

2. Observe the following error response:
```
"json": {
        "code": 500,
        "message": "Command failed: tmsh -a delete ltm data-group internal /Common/appsvcs/settings\n01020036:3: The requested value list (/Common/appsvcs/settings) was not found.\n"
}
```

### Expected Behavior
I would expect either from these two behaviors:
- the data-group AND the API settings both synchronize to the HA peer
- the data-group AND the API settings are both local only, don't synchronize with config

### Actual Behavior
Described in the Summary above

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added this issue to our internal product backlog as AUTOTOOL-2853.

**Comment by seattlevine** [comment]
We see that there is a disconnect here. One solution is to update the internal settings used for the api with the updated settings from the data-group when they are modified. In this case, you would only send settings to one of the devices, not both. Is this acceptable for your use case?

**Comment by m-kratochvil** [comment]
@seattlevine Yes, I would say this would be acceptable. I would update the `settings` api endpoint only on one of the devices from the device group, this would propagate into the internal data-group, that data-group would synchronize to the peer device, and the `settings` api endpoint on the peer device would get updated based on the change in the data-group.

Is my understanding correct?

**Comment by seattlevine** [comment]
Yes, that is correct.

**Comment by m-kratochvil** [comment]
Ok, thanks. I would like to give it another round of testing, because I'm fairly sure the last step doesn't work without a restart of `restnoded` on the peer device.

**Comment by seattlevine** [comment]
Sorry for the confusion. That's what we are fixing. The data group is now read at the start of every request instead of relying on the cached settings.

**Comment by dstokesf5** [comment]
This is resolved in the 3.34.0 release that is now available.
